### PR TITLE
Swap creation fix

### PR DESCRIPTION
**Pull request by monokh**
### Description

Small fixes to make the initiation of swap work

### Parts

- [x] Removed hex validation for generate swap - only ethereum addresses are in hex
- [x] Fixed example

### CODE REVIEW

File: examples/browser/bitcoin/ledger-send-transaction.html [hunk 1]
```diff
@@ -57,7 +57,7 @@ <h3>Pay to custom script</h3>
       $('#sendScript').click(() => {
         // pay to script:
         // null, value, script
-        bitcoin.sendTransaction(null, $('#scriptValue').val(), parseInt($('#toScript').val())).then(result => {
+        bitcoin.sendTransaction(null, parseInt($('#scriptValue').val()), $('#toScript').val()).then(result => {
```
Comment (harshjv) [inline examples/browser/bitcoin/ledger-send-transaction.html:60]: this will fail for floats. e.g. 0.1 ETH

File: src/Client.js [hunk 2]
```diff
@@ -460,14 +460,6 @@ export default class Client {
       throw new TypeError('Secret hash should be a string')
     }
 
-    if (!(/^[A-Fa-f0-9]+$/.test(recipientAddress))) {
```
Comment (harshjv) [inline src/Client.js:463]: nice catch!

File: src/providers/bitcoin/BitcoinLedgerProvider.js [hunk 3]
```diff
@@ -216,10 +216,10 @@ export default class BitcoinLedgerProvider extends Provider {
     await this._connectToLedger()
 
     let receivingAddress
-    if (to == null) {
+    if (data) {
```
Comment (harshjv) [inline src/providers/bitcoin/BitcoinLedgerProvider.js:219]: Nice catch, again!

File: src/providers/bitcoin/BitcoinUtil.js [hunk 4]
```diff
@@ -30,7 +30,7 @@ function compressPubKey (pubKey) {
 function pubKeyToAddress (pubKey, network, type) {
   pubKey = ensureBuffer(pubKey)
   const pubKeyHash = hash160(pubKey)
-  const addr = this.pubKeyHashToAddress(pubKeyHash, network, type)
+  const addr = pubKeyHashToAddress(pubKeyHash, network, type)
```
Comment (harshjv) [inline src/providers/bitcoin/BitcoinUtil.js:33]: Another one! Thanks Mo!

File: examples/browser/bitcoin/ledger-send-transaction.html (hunk 1 above)
Comment (harshjv) [reply examples/browser/bitcoin/ledger-send-transaction.html:60]: you can change it to `parseFloat` or use `Number`

**Review by harshjv — CHANGES_REQUESTED** [review]
Just a small modification required! Other than that, LGTM

File: examples/browser/bitcoin/ledger-send-transaction.html (hunk 1 above)
Comment (monokh) [reply examples/browser/bitcoin/ledger-send-transaction.html:60]: I believe the interface for `sendTransaction` accepts the smallest denomination of the amount. so 1 here would be 1 satoshi or wei. So I think we shouldn't be parsing as floats for the example unless we were doing some conversion from the various denominations

**Comment by monokh** [comment]
comments have been addressed and don't see anything really contentious here so merging

**Pull request merged**